app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    people_instances = [
+        Person(human.get("name"), human.get("age"))
+        for human in people
+    ]
+    for i, person in enumerate(people):
+        if person.get("wife"):
+            people_instances[i].wife = Person.people[person.get("wife")]
```

Comment:
You could use a walrus operator to not retrieve the value by key twice:
```suggestion
        if wife_name := person.get("wife"):
            people_instances[i].wife = Person.people[wife_name]
```